File: EduLink.Domain/Strategies/ImpuestoIncluidoStrategy.cs
using EduLink.Domain.Interfaces;

namespace EduLink.Domain.Strategies;

public class ImpuestoIncluidoStrategy : IPrecioStrategy
{
    private readonly decimal _tasaImpuesto;

    public ImpuestoIncluidoStrategy(decimal tasaImpuesto = 0.12m)
    {
        _tasaImpuesto = tasaImpuesto; // 12% por defecto correa
    }

    public decimal Calcular(decimal precioBase, Cliente cliente)
    {
        return precioBase * (1 + _tasaImpuesto);
    }
}

File: app/domain/strategies/impuesto_incluido_strategy.rb
require 'bigdecimal'
require_relative '../interfaces/i_precio_strategy'

module Strategies
  class ImpuestoIncluidoStrategy
    include Interfaces::IPrecioStrategy

    def initialize(tasa_impuesto = 0.12)
      # Usa AppConfig o el defecto pasado en el constructor
      tasa = tasa_impuesto || AppConfig.tasa_impuesto
      @tasa_impuesto = BigDecimal(tasa.to_s)
    end

    def calcular(precio_base, cliente)
      BigDecimal(precio_base.to_s) * (1 + @tasa_impuesto)
    end
  end
end
